### ISSUE TITLE
`interpolate: false,` settings of sources no longer respected in `ol/source/Raster`

### ISSUE THREAD

**Issue by mike-000** [openlayers/openlayers#14415]
**Describe the bug**
The `interpolate: false,` settings of sources are no longer respected in `ol/source/Raster`.

**To Reproduce**
In v7.1 the `interpolate: false` setting of the source within the RasterSource was always respected as the setting was used to render the source tiles to the view resolutions used by the RasterSource https://codesandbox.io/s/rastersource-7-1-0-kiiw3e

In 7.2 that only happens if `resolutions: null` is specified.  Otherwise the interpolate setting of the source becomes irrelevant as there is no scaling of the tile within the RasterSource, and it is not used when the RasterSource output is rendered to the map https://codesandbox.io/s/rastersource-7-1-0-kiiw3e

Also, regardless of what was specified. unless `resolutions: null` is specified `getData()` now always returns uninterpolated results.

**Expected behavior**
The interpolate setting of the source now also needs to be used when rendering the RasterSource output.  If the RasterSource has multiple sources with differing interpolate settings there would be a conflict which could be avoided by specifying `resolutions: null`.  The documentation for `resolutions` should mention that.

**Comment by tschaub** [comment]
The two links you provide above are for the same sandbox (https://codesandbox.io/s/rastersource-7-1-0-kiiw3e).  Did you mean to provide one example that works as expected and another that does not?

**Comment by mike-000** [comment]
Yes, the second should have been https://codesandbox.io/s/rastersource-7-2-2-coowii - same code but with the OL version updated
